src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ResultObject = [];
+
+  for (const operation of actions) {
+    const CurrentObject = {};
+
+    if (ResultObject.length !== 0) {
+      Object.assign(CurrentObject, ResultObject[ResultObject.length - 1]);
+    }
+
+    if (ResultObject.length === 0) {
+      Object.assign(CurrentObject, state);
+    }
```

Comment:
instead of doing it copy the initial state into `currentState` object before the loop
and before pushing it to the array make a copy